File: src/com/bridgelabz/functional/Leapyear.java
/**
 * @purpose	:Check whether year is Leap Year or not
 * @author user:sangita awaghad
 * @since	:30-07-2019
 */
package com.bridgelabz.functional;

import com.bridgelabz.utility.Utility;

public class Leapyear {
	public static void main(String args[]) {
		Utility utility = new Utility();
		
		int mYear;
		
		do {
			System.out.println("Enter the for digit Year");
			mYear = utility.getInteger();
		} while (!utility.isFourDigit(mYear));

		boolean result = utility.checkerLeapYear(mYear);

		if (result)
			System.out.println("Year " + mYear + " is Leap Year");
		else
			System.out.println("Year " + mYear + " is not Leap Year");

	}
}

File: nodeProject/LeapYear.js
const utility = require("./Utility/Utility");
do {
    console.log("Enter Year");
    var year = utility.getInputNumber();
    var temp = year;
    var count = 0;
    while (temp >= 1) {
        count = count + 1;
        temp /= 10;

    }
    console.log(count);
    if (count != 4)
        console.log("year must be four digit");
} while (count != 4);

var isLeap = utility.isLeapYear(year);
if (isLeap)
    console.log(`${year} is leap year`);
else
    console.log(`${year} is NOT leap year`)